### ISSUE TITLE
[Feature Request] Help Us TK-Glitch, You're Our Only Hope! Doom Eternal requires certain extensions not yet implemented in Proton Wine

### ISSUE THREAD

**Issue by gardotd426** [Tk-Glitch/PKGBUILDS#520]
So, Doom Eternal doesn't work, because of this whole thing:

https://github.com/ValveSoftware/wine/pull/85

Doom Eternal is apparently one of the games that requires those new extensions/features. However, someone was able to get it working (with a big asterisk) by using one of your patches, doing some other trickery, and turning off internet (the asterisk in question is they have a DRM-free copy). 

Others have had trouble getting Wine to build with the patch. Is there anything you might be able to help out with? I imagine you don't have the game, but if you could look over the issues page, I know you and GE are miracle workers. 

https://github.com/ValveSoftware/Proton/issues/3654

**Comment by ronoverdrive** [comment]
Without build logs you're asking him to go in blind by saying people are having trouble building with the patch. If you're having problems please provide your logs so the problem can be narrowed down.

**Comment by Tk-Glitch** [comment]
Ok, so, after adding the required change with https://github.com/Tk-Glitch/PKGBUILDS/commit/52d0e915a7c88acaf544ad1c9b7eb9fcc23645cb yesterday and getting confirmations about it working all fine, the patchsets are now part of proton-tkg by default and can be optionally enabled on wine-tkg with `_protonify="true"` as of https://github.com/Tk-Glitch/PKGBUILDS/commit/08a666bea87a08a7ac8e8a265033ef41be2d3d61 .

The existing `winevulkan_fshack_opts.mypatch` community patch is back to its former state and shouldn't be required for Doom Eternal to work anymore.

Sidenotes:
- For AMD GPU users RADV/ACO doesn't allow the game to run, RADV/LLVM and AMDVLK are running terribly, while AMDGPU-PRO seemingly runs fine.
- The game reportedly works with the denuvo protected executable but it seems to require either `detroit_BH.mypatch` or `FinalFantasyXVHack.mypatch` community patches in addition, and this is the reason why it doesn't work with vanilla Proton 5.0-5.

I would assume the issue has been addressed 🐸

**Comment by gardotd426** [comment]
As always, you're a gem. Weird that it needs the D:BH or FFXV patches (I
know those use Denuvo but still), but yeah I was waiting to buy until we
knew if it was gonna work or not, luckily as you may remember I have a Navi
card (it seems GTX is performing pretty crappily with this game right now)
and yeah I installed the 19.50 amdgpu-pro vulkan drivers and all that. But
hey, while I have you here, what's up with the note in the README for the
patches saying that RDR2 needs that patch, does that mean RDR2 is also on
the verge of working? Or is there some sort of news?

**Comment by Tk-Glitch** [comment]
> does that mean RDR2 is also on the verge of working

Well, yes. Brendan has been hard at work, and with a combination of patches and hacks you can get ingame. But it'll need a bit more to be actually usable.

**Comment by gardotd426** [comment]
That's understandable, it had just always seemed like that was one game that wasn't even close to working. I just thought about something with Doom Eternal though before I go ahead and close this. For some reason even after updating all the available vulkan packages to 1.2.135, is there some reason why `vulkaninfo` and the steam system information thing still show older vulkan api versions? Like, here's my `vulkaninfo | grep apiVersion` for my RX 5600 XT:
```
vulkaninfo | grep apiVersion
ERROR: [Loader Message] Code 0 : /usr/lib32/libvulkan_intel.so: wrong ELF class: ELFCLASS32
ERROR: [Loader Message] Code 0 : /usr/lib32/amdvlk32.so: wrong ELF class: ELFCLASS32
ERROR: [Loader Message] Code 0 : /usr/lib32/libvulkan_radeon.so: wrong ELF class: ELFCLASS32
ERROR: [Loader Message] Code 0 : /opt/amdgpu-pro/lib/i386-linux-gnu/amdvlk32.so: wrong ELF class: ELFCLASS32
ERROR: [Loader Message] Code 0 : /usr/lib32/amdvlk32.so: wrong ELF class: ELFCLASS32
	apiVersion     = 4202629 (1.2.133)
	apiVersion     = 4202629 (1.2.133)
	apiVersion     = 4198529 (1.1.129)
	apiVersion     = 4202624 (1.2.128)
```
I have RADV, AMDVLK, and vulkan-amdgpu-pro, which is why there's 4, FOSS AMDVLK is the two 1.2.133's, `vulkan-amdgpu-pro` is the 1.1.129, RADV is the 1.2.128. But then the `Vulkan Instance Version` is 1.2.135, as one would expect. What's that about? It seems like all the screenshots with the performance stats turned on for DE show that Vulkan is some pre-1.2.135 version as well, so I assume it doesn't mean anything's wrong, does your Navi card show the same goofiness? Cause it wasn't just `vulkan-icd-loader` and `vulkan-headers` that got an update earlier, it was like the whole vulkan stack, vulkan tools (even mesa vulkan, `vulkan-radeon` got updated earlier). I'll go ahead and close the issue since you've actually fixed the original request, just feel free to answer if you want whenever you have time. Thanks man, you're a hero for real.

**Comment by Tk-Glitch** [comment]
You can ignore that indeed. It'll report whatever is in the icd json file. You could set it to e.g. 9.9.99 in there if you wanted to get some interesting screenshots. Nothing to worry about 🐸
